### DISCUSSION TITLE
I hate that there is no default routing setup

### DISCUSSION THREAD

**Discussion by idc77** [vuejs/router#2271] [Ideas]
I read the documentation, 1st of all there's no import in your getting started.
There is no route dir and default definitions like there is with quasar.
I have to manually, time intensively set up the routing
There is not init script to simple create a default / router and a 404 catchall.

Add some comfort, provide at least the basics.

For instance this is the quasar src/router/index.js
```js
import { route } from 'quasar/wrappers'
import { createRouter, createMemoryHistory, createWebHistory, createWebHashHistory } from 'vue-router'
import routes from './routes'

export default route(function (/* { store, ssrContext } */) {
  const createHistory = process.env.SERVER
    ? createMemoryHistory
    : (process.env.VUE_ROUTER_MODE === 'history' ? createWebHistory : createWebHashHistory)

  const Router = createRouter({
    scrollBehavior: () => ({ left: 0, top: 0 }),
    routes,

    // Leave this as is and make changes in quasar.conf.js instead!
    // quasar.conf.js -> build -> vueRouterMode
    // quasar.conf.js -> build -> publicPath
    history: createHistory(process.env.VUE_ROUTER_BASE)
  })

  return Router
})
```

and then there's 
routes.js
```js
const routes = [
  {
    path: '/',
    component: () => import('layouts/MainLayout.vue'),
    children: [
      { path: '', name: 'index', component: () => import('pages/IndexPage.vue') }
    ]
  },

  // Always leave this as last one,
  // but you can also remove it
  {
    path: '/:catchAll(.*)*',
    component: () => import('pages/ErrorNotFound.vue')
  }
]

export default routes
```

Why can't something like this be done with some kind of init script?
"Would you like to create a default routing skeleton? [Y/n]"

**Comment by posva** [comment]
You must have been reading the wrong version of the docs. These are the correct ones: https://router.vuejs.org/installation.html and they include the imports and the scaffold instructions.

![Screenshot 2024-06-17 at 09 18 40](https://github.com/vuejs/router/assets/664177/6113588d-c9fc-4762-9c3d-eacf7fc00df0)


Just so you know, bringing negativity to an open-source project like you just did will rarely be helpful (or welcomed). Rather than writing a rant, an objective list of pain points would be more welcome. Cheers.
